app/models/user_role.rb

```diff
@@ -11,6 +11,18 @@ class UserRole < ActiveRecord::Base
   FACILITY_STAFF = "Facility Staff".freeze
   FACILITY_SENIOR_STAFF = "Facility Senior Staff".freeze
 
+  module AssociationExtension
+
+    def operator?(facility)
+      any? { |ur| ur.facility == facility && ur.in?(facility_roles) }
+    end
+
+    def manager?(facility)
+      any? { |ur| ur.facility == facility && ur.in?(facility_management_roles) }
+    end
+
```

Comment:
I'm sure this works, and there's no real need to throw more time into it...

But, it's not my favorite Rails construct and I wonder if the problem is the `any?` and you could alternately compose scopes:

Feel perfectly free to ignore as the ravings of a madman.

```
def self.in_facility(facility)
  where(facility: facility) 
end

def self.one_of(roles)
  where(role: roles)
end

def self.operator?(facility)
  in_facility(facility).one_of(facility_roles)
end

def self.manager?(facility)
  in_facility(facility).one_of(facility_managment_roles)
```